internal/cli/filtering.go

```diff
@@ -19,7 +18,7 @@ type Filtering struct {
 
 	Interactive    bool     `kong:"group='filter',short='i',help='Select notes interactively with fzf.'" json:"-"`
 	Limit          int      `kong:"group='filter',short='n',placeholder='COUNT',help='Limit the number of notes found.'" json:"limit"`
-	Match          string   `kong:"group='filter',short='m',placeholder='QUERY',help='Terms to search for in the notes.'" json:"match"`
+	Match          []string `kong:"group='filter',short='m',placeholder='QUERY',help='Terms to search for in the notes. If given multiple times each argument is joined with a boolean AND.'" json:"match"`
```

Comment:
This is true for most other options, so I don't think it's necessary to mention it. It's also described by Kong in the flag name:

```
-m, --match=QUERY,...
```
```suggestion
	Match          []string `kong:"group='filter',short='m',placeholder='QUERY',help='Terms to search for in the notes."`
```